docs/docs/migration/v0.0.56-RC6-to-v0.0.57-0.md

```diff
@@ -0,0 +1,156 @@
+-   Make the saved file excutable: `chmod +x ~/helm_deployment_migration.sh`
+-   Run helm_deployment_migration.sh:
+
+```bash
+# Go to the Home Directory where you saved the helm_deployment_migration.sh script
+cd ~
+# helm_deployment_migration.sh script.
+# *NOTE* Here `magda` is your helm release name.
+# Please replace it with your actual helm release name if you use a different name.
+# If you use Magda's Terraform module, the default release name is `magda`
+./helm_deployment_migration.sh magda
+```
+
+Once the script starts to run, it will show you a `git diff` screen to highlight the helm template changes that it's about to make. You can type `q` to quit the review screen and press `enter` to continue with the changes (or `ctrl-c` to abort). After that, you need to press `enter` (or `ctrl-c` to abort) to proceed with applying the change to the release data in cluster. Once it's done, the current release manifest will be fetched again and display on the screen. You will find the release only contains new k8s APIs usage now.
+
+#### Step 2: Install [Helm 3](https://helm.sh/docs/intro/install/)
+
+Follow the instruction [here](https://helm.sh/docs/intro/install/) to install helm3.
+
+You can also install helm 3 with brew:
+
+```bash
+brew install helm@3
+```
+
+#### Step 3: Install [Helm 2to3 Plugin](https://github.com/helm/helm-2to3)
+
+Follow the instruction here to install Helm 2to3 Plugin:
+
+https://github.com/helm/helm-2to3#install
+
+#### Step 4: Migrate Helm v2 configuration
+
+Follow the `Migrate Helm v2 configuration` instruction here:
+
+https://github.com/helm/helm-2to3#migrate-helm-v2-configuration
+
+#### Step 5: Convert Helm 2 Release data to Helm 3 release data in your cluster
+
+Run the following to migrate Helm v2 release data to Helm 3
+
+```bash
+# Here `magda` is the helm release name
+helm 2to3 convert magda --delete-v2-releases
+```
+
+> NOTE: --delete-v2-releases switch will delete the v2 release data after the migration. If you don't specify this switch, you will have to use `helm 2to3 cleanup --release-cleanup` to delete helm v2 release data for your release later. `helm 2to3 cleanup` command will remove all helm v2 tiller release data and you can't specify a single release name to remove.
+
+#### Step 6: Upgrade to v0.0.57-0 using helm3 upgrade
+
+Before upgrade to `v0.0.57-0`, run `helm repo list` to make sure the repo `magda-io` (url: https://charts.magda.io) and the repo stable (url: https://kubernetes-charts.storage.googleapis.com) are included. If not, run
+
+```bash
+helm repo add magda-io https://charts.magda.io
+helm repo add stable https://kubernetes-charts.storage.googleapis.com
+```
+
+to add the Magda helm chart repo.
+
+Run `helm upgrade` to upgrade your release:
+
+```bash
+helm --namespace [your magda deploy namespace] upgrade --install --timeout 9999s --wait -f [path to your config value file] magda magda --repo https://charts.magda.io --version 0.0.57-0
+```
```

Comment:
As discussed, I'm getting 

```
Error: UPGRADE FAILED: rendered manifests contain a new resource that already exists. Unable to continue with update: existing resource conflict: namespace: default, name: es-ds, existing_kind: apps/v1, Kind=DaemonSet, new_kind: apps/v1, Kind=DaemonSet
```
:(

Reply:
I missed one step:
```bash
kubectl apply -f https://raw.githubusercontent.com/magda-io/magda/master/deploy/helm/openfaas/crds/crd.yaml
```
in the doc --- have added that step in.
If your cluster already in error status, you can try helm v3.2.0 as it handles errors status better.